drivers/fxos8700/fxos8700.c

```diff
@@ -0,0 +1,205 @@
+
+    i2c_acquire(dev->p.i2c);
+    if(i2c_init_master(dev->p.i2c, I2C_SPEED) != 0) {
+        DEBUG("[fxos8700] Can't initialize I2C master\n");
+        i2c_release(dev->p.i2c);
+        return FXOS8700_NOBUS;
+    }
+
+    if (i2c_read_regs(dev->p.i2c, dev->p.addr, FXOS8700_REG_WHO_AM_I, &config, 1) <= 0) {
+        i2c_release(dev->p.i2c);
+        DEBUG("[fxos8700] Could not read WHOAMI\n");
+        return FXOS8700_NOBUS;
+    }
+    if (config != FXOS8700_WHO_AM_I_VAL) {
+        i2c_release(dev->p.i2c);
+        DEBUG("[fxos8700] WHOAMI is wrong (%2x)\n", config);
+        return FXOS8700_NOBUS;
+    }
+
+    /* Configure the ODR to maximum (400Hz in hybrid mode) */
+    config = 0x00;
+    if (fxos8700_write_regs(dev, FXOS8700_REG_CTRL_REG1, &config, 1) != FXOS8700_OK) {
+        return FXOS8700_NOBUS;
+    }
+    /* Activate hybrid mode */
+    config = HYBRID_MODE;
+    if (fxos8700_write_regs(dev, FXOS8700_REG_M_CTRL_REG1, &config, 1) != FXOS8700_OK) {
+        return FXOS8700_NOBUS;
+    }
+    /* Set burst read mode (accel + magnet together) */
+    config = 0x20;
+    if (fxos8700_write_regs(dev, FXOS8700_REG_M_CTRL_REG2, &config, 1) != FXOS8700_OK) {
+        return FXOS8700_NOBUS;
+    }
+
+    /* initial read for caching operation */
+    if (fxos8700_read(dev, &acc_cached, &mag_cached) != FXOS8700_OK) {
+        return FXOS8700_BUSERR;
+    }
+    last_read_time = xtimer_now_usec();
+
+    return FXOS8700_OK;
+}
+
+
+int fxos8700_set_active(fxos8700_t* dev)
+{
+    uint8_t config = 0x01;
+    if (fxos8700_write_regs(dev, FXOS8700_REG_CTRL_REG1, &config, 1) != FXOS8700_OK) {
+        return FXOS8700_BUSERR;
+    }
+    return FXOS8700_OK;
+}
+
+int fxos8700_set_idle(fxos8700_t* dev)
+{
+    uint8_t config = 0x00;
+    if (fxos8700_write_regs(dev, FXOS8700_REG_CTRL_REG1, &config, 1) != FXOS8700_OK) {
+        return FXOS8700_BUSERR;
+    }
+    return FXOS8700_OK;
+}
+
+int fxos8700_read(const fxos8700_t* dev, fxos8700_measurement_t* acc, fxos8700_measurement_t* mag)
+{
+    uint8_t data[12];
+    uint8_t ready = 0;
+
+    if (fxos8700_set_active(dev)) {
+        return FXOS8700_BUSERR;
+    }
+
+    while(!(ready & 0x08)) {
+        fxos8700_read_regs(dev, FXOS8700_REG__STATUS, &ready, 1);
+    }
+    while(!(ready & 0x08)) {
+        fxos8700_read_regs(dev, FXOS8700_REG_M_DR_STATUS, &ready, 1);
+    }
+
+    /* Read all data at once */
+    if (fxos8700_read_regs(dev, FXOS8700_REG_OUT_X_MSB, &data[0], 12)) {
+        return FXOS8700_BUSERR;
+    }
+
+    if (fxos8700_set_idle(dev)) {
+        return FXOS8700_BUSERR;
+    }
+
+    /* Read accelerometer */
+    if (acc) {
+        int32_t acc_raw_x = (int32_t) ((data[0]<<6) | (data[1]>>2));
+        int32_t acc_raw_y = (int32_t) ((data[2]<<6) | (data[3]>>2));
+        int32_t acc_raw_z = (int32_t) ((data[4]<<6) | (data[5]>>2));
+        acc->x = (int16_t)(acc_raw_x*244/1000);
+        acc->y = (int16_t)(acc_raw_y*244/1000);
+        acc->z = (int16_t)(acc_raw_z*244/1000);
```

Comment:
Code style: add spaces around operators

Reply:
Also, for clarity it may be better to have parentheses around `(raw * 244) / 1000`, though it changes nothing in the generated code since the associativity is left to right for the multiplication and division operators.